src/pyhf/readxml.py

```diff
@@ -48,12 +48,11 @@ def import_root_histogram(rootdir, filename, path, name, filecache=None):
         h = f[name]
     except KeyError:
         try:
-            h = f[os.path.join(path, name)]
+            h = f[str(Path(path).joinpath(name))]
         except KeyError:
             raise KeyError(
-                'Both {0:s} and {1:s} were tried and not found in {2:s}'.format(
-                    name, os.path.join(path, name), os.path.join(rootdir, filename)
-                )
+                f'Both {name} and {Path(path).joinpath(name)} were tried and not found'
+                f' in {Path(rootdir).joinpath(filename)}'
```

Comment:
This try except block is now not getting covered in the tests for reasons that aren't clear to me: https://codecov.io/gh/scikit-hep/pyhf/pull/798/diff?src=pr&el=tree#D3-49

It would be good to fix these if possible.

Reply:
I have restructured 54-55 into a single line for now and would probably revert to using os.path if it still does not get covered in the tests.